### ISSUE TITLE
set doesn't work with array

### ISSUE THREAD

**Issue by impredicative** [fboender/jsonxs#2]
```
import jsonxs

d={'a': [{'e': 'f'}, {'x': 'y'}]}

jsonxs.jsonxs(d, 'a[0].e')
'f'

jsonxs.jsonxs(d, 'a[0].e', jsonxs.ACTION_SET, 'q')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "python3.4/site-packages/jsonxs/jsonxs.py", line 168, in jsonxs
    prev_path[token] = value
TypeError: list indices must be integers, not str
```

**Comment by impredicative** [comment]
I would love it if there is a simple fix for this today since my project is blocked otherwise.

**Comment by fboender** [comment]
Fixed in master and v0.6 on PyPi

**Comment by impredicative** [comment]
0.6 is working well. Thanks very much.
